html/src/main/scala/org/hyperscala/html/Position.scala

```diff
@@ -0,0 +1,51 @@
+package org.hyperscala.html
+
+/**
+ * Get a [[scala.collection.Seq]] with a value if in a matching location, as would be useful
+ * for site navigation.
+ *
+ * ==Example==
+ *
+ * If we want `"active"` in the `clazz` [[scala.collection.Seq]] when in the matching location,
+ * one could do the following.
+ *
+ * {{{
+ * class BasePage extends Webpage(MySite) with Position[String, String] {
+ *
+ *   //Create partially applied function to simplify use
+ *   val inPos = inPosition(List.empty, _:String, "active")
+ *
+ *   ...  //implement some common navigation here in Parent class
+ *
+ *         contents += new tag.Li(clazz = inPos("index")) {
+ *           contents += new tag.A(href = "index.html") {
+ *             contents += "Home"
+ *           }
+ *         }
+ *
+ * }
+ * }}}
+ *
+ * Then in an `IndexPage` page that extends `BasePage`, set the `posId` to `"index"`
+ *
+ * {{{
+ * object IndexPage extends BasePage {
+ *   override def posId: Option[String] = Some("index")
+ * }}}
+ *
+ *  The result will then be that the navigation built for the `IndexPage` will have `"active"` applied to the
+ *  list item tested with `inPos("index")`.
+ *
+ * `inPos("index")` would return `List("active")`
+ *
+ *
+ * @tparam A The type for the position identifier
+ * @tparam B The type for the elements of the Seq
+ */
+trait Position[A,B] {
```

Comment:
The idea is that you could have a base page with something like navigation, then each derived page would inherit that navigation, but each derived page would have its own distinct _Position_.  This makes it simple to have elements in the navigation become **active** when the navigation element corresponds to its matching _Position_.  An example would be that when on a _Home_ page the _home_ navigation element would/could be made to look and or work a special way because one is _on the home page_.  

Here is what one could add to a base/parent class.

```scala
class BasePage extends Webpage(HelloSite) with Position[String, String] {
 
  override def posId: Option[String] = None
  val isPos = inPosition(List.empty, _:String, "active")
```

Then some navigation could be created using this . . . 

```scala
contents += new tag.Div(id = "bs-site-navbar-collapse-1", clazz = List("collapse", "navbar-collapse")) {
        contents += new tag.Ul(clazz = List("nav", "navbar-nav", "navbar-right")) {
          contents += new tag.Li(clazz = isPos("index")) {
            contents += new tag.A(href = "index.html") {
              contents += "Home"
            }
          }
```
So if a derived class/page has a _Position_ identified by `index`, then the above `li` tag will have `active` assigned where the `isPos("index")` is shown.

The index page would set its _Position_ like this.

```scala
  override def posId: Option[String] = Some("index")
```

I did a test app where I altered the navigation this way to help the user see which page they loaded.